openedx/openedx-platform
# feat: update search index when course content is updated [FC-0040]

## Description
This PR implements updating the search index with content metadata whenever they change

## More Infomation
- Closes https://github.com/openedx/modular-learning/issues/196
- Depends on:
  - https://github.com/openedx/edx-platform/pull/34310

## Testing Instructions
- Run your local stack on this branch
- Make sure you have `meilisearch` setup locally, follow the setup instructions here https://github.com/open-craft/tutor-contrib-meilisearch
- Make sure you also have some sample taxonomies/tags, you can add some from here: https://github.com/open-craft/taxonomy-sample-data
- Run `tutor dev run cms bash` and `./manage.py cms reindex_studio`
- View the resulting index at http://meilisearch.local.edly.io:7700/ (see [tutor-contrib-meilisearch README](https://github.com/open-craft/tutor-contrib-meilisearch) for how to get the API key to log in)
- Go back to studio, and add/update XBlocks from a course
- Go back to meilisearch dashboard, and confirm that the XBlocks docs have been updated to reflect the changes you made
- Repeat and confirm the above for both units and blocks
- Also, try to create a Content Library with some components and check if the index is updated accordingly.

**NOTE:** Meilisearch seems to cache queries along with their results in the frontend, so if you simply search it might 
show you stale data (network tab shows no requests), especially if you're searching for the same query. Make sure you refresh the Meilisearch dashboard (http://meilisearch.local.edly.io:7700/) and then perform the search.

___
Private ref: [FAL-3690](https://tasks.opencraft.com/browse/FAL-3690)
